### PR TITLE
PMP: repair polygon soup

### LINKED ISSUE #1152: degenerate faces in a polygon soup
I get a segfault with this [program](https://gist.github.com/afabri/f1fa1552e8421c95d35a52f471429d35).
It has a face with three identical vertex indices (as it may happen when reading STL data).

### LINKED ISSUE #3231: Add a function to merge duplicated points in polygon soups
See the code here
https://gist.github.com/sloriot/12f6ba2c5d593b1b0e2e82a07fc82df7

### PR DESCRIPTION
**Pull request by MaelRL**
## Summary of Changes

This PR adds a few functions to fix incorrect polygon soups and some concept links fixes.

## Release Management
#3201 must be integrated first.
* Affected package(s):  `Polygon_mesh_processing`
* Issue(s) solved (if any): fix #1152, fix #3231 
* Feature/Small Feature (if any): [PMP_polygon_soup_repair](https://cgal.geometryfactory.com/CGAL/Members/wiki/Features/Small_Features/PMP_polygon_soup_repair) - pre-approve 2018/09/05

### CODE REVIEW

File: Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/repair_polygon_soup.h [hunk 1]
```diff
@@ -0,0 +1,440 @@
...
+    if(simplify_polygon(points, polygon, traits))
+      ++simplified_polygons_n;
+  }
+
+  return simplified_polygons_n;
+}
+
+// \ingroup PMP_repairing_grp
+//
+// Splits "pinched" poygons, that is polygons for which a point appears more than once,
```
Comment (sloriot) [inline Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/repair_polygon_soup.h]: poygons

File: Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/repair_polygon_soup.h [hunk 2]
```diff
@@ -0,0 +1,440 @@
...
+    Unique_point_container unique_points(traits.less_xyz_3_object());
+
+    for(std::size_t i=0, polygon_size = polygon.size(); i<polygon_size; ++i)
+    {
+      const Point_3& p = points[polygon[i]];
+
+      std::pair<typename Unique_point_container::iterator, bool> is_insert_succesful =
+        unique_points.insert(std::make_pair(p, i));
+
+      if(!is_insert_succesful.second)
```
Comment (sloriot) [inline Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/repair_polygon_soup.h]: succesful

**Comment by sloriot** [comment]
@MaelRL could you please solve the conflicts and fix the TODOs, I would like this branch to be integrated next week.

**Comment by lrineau** [comment]
Just for info: the conflicts are still not solved. I cannot test the PR tonight.

**Comment by MaelRL** [comment]
Needed merge by #3201 first, should be good to test now.

**Comment by lrineau** [comment]
`CHANGES.md` seems unchanged.

**Comment by sloriot** [comment]
Successfully tested in [4.14-Ic-32](https://cgal.geometryfactory.com/CGAL/testsuite/results-4.14-Ic-32.shtml).

**Pull request merged**